pkg/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/pulumi/pulumi/pkg/v3
 
-go 1.22
+go 1.24
```

Comment:
I don't think we can do this yet. We want to allow 1.23 still.

Maybe there's a case to be made for allowing `pkg` be 1.24 (and keeping `sdk` at 1.23), but that would very likely add a bunch of CI complexity around the testing matrix.